core/src/main/java/org/apache/calcite/plan/RelOptCluster.java

```diff
@@ -146,18 +146,26 @@ public RexBuilder getRexBuilder() {
    * @param metadataProvider custom provider
    */
   @EnsuresNonNull({"this.metadataProvider", "this.metadataFactory"})
+  @SuppressWarnings("deprecation")
   public void setMetadataProvider(
       @UnknownInitialization RelOptCluster this,
       RelMetadataProvider metadataProvider) {
     this.metadataProvider = metadataProvider;
-    this.metadataFactory = new MetadataFactoryImpl(metadataProvider);
+    this.metadataFactory =
+        new org.apache.calcite.rel.metadata.MetadataFactoryImpl(metadataProvider);
```

Comment:
This change does not seem necessary? Could we leave it unmodified?

Reply:
The deprecation test will fail if imports are used for a deprecated class.